packages/interactivity-router/src/assets/test/styles.test.ts

```diff
@@ -0,0 +1,561 @@
+
+		expect( attributes ).toEqual( [
+			{
+				id: 'style1',
+				media: null,
+				originalMedia: null,
+			},
+			{
+				id: 'style1Media',
+				media: 'screen',
+				originalMedia: null,
+			},
+			{
+				id: 'link1',
+				media: null,
+				originalMedia: null,
+			},
+			{
+				id: 'link1Media',
+				media: 'screen',
+				originalMedia: null,
+			},
+			{
+				id: 'style2',
+				media: 'preload',
+				originalMedia: null,
+			},
+			{
+				id: 'style2Media',
+				media: 'preload',
+				originalMedia: 'screen',
+			},
+			{
+				id: 'link2',
+				media: 'preload',
+				originalMedia: null,
+			},
+			{
+				id: 'link2Media',
+				media: 'preload',
+				originalMedia: 'screen',
+			},
+		] );
+	} );
+} );
+
+// Tests for prepareStyles function
+describe( 'prepareStyles', () => {
+	beforeEach( () => {
+		document.head.replaceChildren();
+	} );
+
+	afterAll( () => {
+		document.head.replaceChildren();
+	} );
+
+	it( 'should use cached promises for the same URL', () => {
+		// Create a test document
+		const doc = document.implementation.createHTMLDocument();
+		const style = doc.createElement( 'style' );
+		doc.head.appendChild( style );
+
+		// First call should update the DOM
+		const firstResult = prepareStyles( doc, 'https://example.com/test' );
+		expect( firstResult ).toBeTruthy();
+
+		// Second call should return the same promises
+		const secondResult = prepareStyles( doc, 'https://example.com/test' );
+		expect( secondResult ).toBe( firstResult );
+	} );
+
+	it( 'should extract style elements from the provided document', () => {
+		// Create a test document with style elements
+		const doc = document.implementation.createHTMLDocument();
+		const style1 = doc.createElement( 'style' );
+		style1.id = 'test-style-1';
+		const style2 = doc.createElement( 'link' );
+		style2.id = 'test-link-1';
+		style2.rel = 'stylesheet';
+		doc.head.appendChild( style1 );
+		doc.head.appendChild( style2 );
+
+		prepareStyles( doc );
+
+		// Check that styles were extracted and added to the document
+		expect( document.querySelector( '#test-style-1' ) ).toBeTruthy();
+		expect( document.querySelector( '#test-link-1' ) ).toBeTruthy();
+	} );
+} );
+
+// Tests for applyStyles function
+describe( 'applyStyles', () => {
+	beforeEach( () => {
+		document.head.replaceChildren();
+	} );
+
+	afterAll( () => {
+		document.head.replaceChildren();
+	} );
+
+	it( 'should enable included styles and disable others', () => {
+		// Create some style elements
+		const style1 = createStyleElement( 'style1' );
+		const style2 = createStyleElement( 'style2' );
+		const style3 = createStyleElement( 'style3' );
+
+		// Add all to document
+		document.head.appendChild( style1 );
+		document.head.appendChild( style2 );
+		document.head.appendChild( style3 );
+
+		// Init `sheet` property
+		mockSheet( style1, { disabled: true, mediaText: 'all' } );
+		mockSheet( style2, { disabled: false, mediaText: 'all' } );
+		mockSheet( style3, { disabled: false, mediaText: 'prefetch' } );
```

Comment:
I've noticed that in these tests, you are still using `prefetch` instead of `preload`. Is that on purpose?

```suggestion
		mockSheet( style3, { disabled: false, mediaText: 'preload' } );
```

Reply:
Nope, good catch! Fixed in https://github.com/woocommerce/gutenberg/pull/2/commits/c1cc3266ad73ac860b626c430b64b06358bdf429.